### DISCUSSION TITLE
posts/vue%E5%BC%80%E5%8F%91/vue3-+-vite-+-typescript-+-bootstrap5-%E5%BC%80%E5%8F%91%E6%8C%87%E5%8D%97/

### DISCUSSION THREAD

**Discussion by giscus** [dstweihao/dst-hugo-giscus#7] [Announcements]
# posts/vue%E5%BC%80%E5%8F%91/vue3-+-vite-+-typescript-+-bootstrap5-%E5%BC%80%E5%8F%91%E6%8C%87%E5%8D%97/

前言 因为之前只是用Vue来开发IaaS平台项目，其使用场景通常都是PC，所以没怎么涉及到Web响应式设计开发相关知识。最近项目刚好是应用在PC和平板，甚至是手机上的，所以就需要考虑响应式的问题。
1. 技术选型 Web响应式开发，主要是基于CSS里@media媒体查询实现，允许不同视图尺寸使用不同的布局。
1.1 方案一：Vue 3 + Vite + Bootstrap5 1.1.1 优点 容易上手：只要对 HTML 和 CSS 有基本了解的人都可以很快速的使用 Bootstrap。
响应式设计：Bootstrap 可以根据不同平台（手机、平板电脑和台式机）进行调整。
移动优先：在 Bootstrap 中，自适应移动端是框架的核心部分。
浏览器兼容性：Bootstrap5 兼容所有主流浏览器（Chrome、Firefox、Edge、Safari 和 Opera）。 如果您需要支持 IE11 及以下版本，请使用 Bootstrap4 或 Bootstrap3。
github 星数 164k
Bootstrap5 效果演示：https://www.bootstrap-admin.top/
1.1.2 缺点 没办法使用我们积累的vue next组件库
Bootstrap5 需要学习成本，但不会很难
相对紧急的项目，通常不应该引入新技术去开发实现，而是以快速交付为主。
1.2 方案二：Vue 3 + Vite + Element Plus + @media 1.2.1 优点 基于前端现有技术积累，可快速开发项目功能 1.2.2 缺点 前端封装组件库，并没考虑过响应式设计，潜在未知样式问题，如开发此项目，各组件需考虑各平台（电脑、平板、手机）屏幕适配的同时，还要考虑已引用该库开发的项目兼容性问题。
@media媒体查询需每个页面、每个平台去适配（Bootstrap5做的事情），工作量相对较大，因无更多平台设备去验证，潜在未知的样式问题。
1.3 结论 简单来说，可以理解为方案一Bootstrap5帮我们处理好了屏幕适配的问题，我们只需要关注功能开发即可，方案二则是功能也要开发，样式也要关注。长远来看，倾向于方案一开发。
2. 快速搭建 Bootstrap5官方文档有提供Vite基于开发的指南《Bootstrap & Vite》，我们虽然是基于Vue3开发的，但是相关配置基本都一样的，只是有一点点差异而已。

https://dstweihao.cn/posts/vue%E5%BC%80%E5%8F%91/vue3-+-vite-+-typescript-+-bootstrap5-%E5%BC%80%E5%8F%91%E6%8C%87%E5%8D%97/

**Comment by hanhanbina** [comment]
很乱，很多不能这样写

**Reply by dstweihao** [reply]
这个起初是打算做一个大型项目，结果只需要一个Demo，所以也没特别深入了，感兴趣可以看这个教程，应该更好一些哈。https://juejin.cn/column/6969383549560422414。
